### PR TITLE
[Wav2Vec2FeatureExtractor] Fix `extractor.pad()` dtype backwards compatibility

### LINKED ISSUE #13689: New Wav2Vec2 padding has slightly backward breaking changes
The PR: https://github.com/huggingface/transformers/pull/13650 introduced some quite tricky backwards breaking changes that we should try to fix. 

The problem is the following: A user might directly use `feature_extractor.pad(...)` instead of `feature_extractor(...)` to just pad already preprocessed inputs in, *e.g.* a data collator.

The following code correctly returned `torch.float32` before merging the PR while the new PR returns `torch.float64` which is slighly breaking and can lead to errors in current fine-tuning Wav2Vec2 scripts:

```python
from transformers import Wav2Vec2FeatureExtractor
import numpy as np

extractor = Wav2Vec2FeatureExtractor.from_pretrained("facebook/wav2vec2-base-960h")

rand_input = np.ones((100,), dtype=np.float64)
out = extractor.pad([{"input_values": rand_input}], return_tensors="pt")

print(out.dtype) # <- this should be `torch.float32`
```

Here is a colab showing how the "old" version works correctly: https://colab.research.google.com/drive/10TlRWvwKx34UORmYdCFAyMWKUU3OtPRf?usp=sharing

Here is a colab showing how the "new" version works incorrectly: 
https://colab.research.google.com/drive/1cXGuG4Rnypmivdm-vdE-61BA1f4hC4e8?usp=sharing

### PR DESCRIPTION
**Pull request by anton-l**
Resolves #13689

This fixes an issue introduced by #13650 with speech feature extractors' tensors being returned as torch.float64 when `.pad()` is called directly:

```python
from transformers import Wav2Vec2FeatureExtractor
import numpy as np

extractor = Wav2Vec2FeatureExtractor.from_pretrained("facebook/wav2vec2-base-960h")

rand_input = np.ones((100,), dtype=np.float64)
out = extractor.pad([{"input_values": rand_input}], return_tensors="pt")

print(out.dtype) # <- this should be `torch.float32`
```

This is due to how pytorch converts float numpy arrays (new padding logic) vs python lists (old padding logic):

* uses torch.float32 for python lists by default: `torch.tensor([1.2, 2.3]).dtype  #  torch.float32`
* `np.array([1.2, 2.3]).dtype  # np.float64` 
* uses source dtype for numpy arrays: `torch.tensor(np.array([1.2, 2.3])).dtype  #  torch.float64`

### CODE REVIEW

File: src/transformers/feature_extraction_sequence_utils.py [hunk 1]
```diff
@@ -187,23 +187,6 @@ def pad(
         padding_strategy = self._get_padding_strategies(padding=padding, max_length=max_length)
 
         required_input = processed_features[self.model_input_names[0]]
-        if required_input and not isinstance(required_input[0], np.ndarray):
```
Comment (anton-l) [inline src/transformers/feature_extraction_sequence_utils.py:190]: Leftover logic from the old implementation, `required_input[0]` is always `np.ndarray` now

File: src/transformers/feature_extraction_sequence_utils.py [hunk 2]
```diff
@@ -240,6 +223,8 @@ def pad(
             for key, value in outputs.items():
                 if key not in batch_outputs:
                     batch_outputs[key] = []
+                if value.dtype is np.dtype(np.float64):
+                    value = value.astype(np.float32)
```
Comment (anton-l) [inline src/transformers/feature_extraction_sequence_utils.py:227]: Instead of adding dtype customization to `.pad()`, just always use `np.float32`, until we really need `float64` support for some new models.

File: src/transformers/feature_extraction_sequence_utils.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/transformers/feature_extraction_sequence_utils.py:190]: good catch!

**Review by patrickvonplaten — APPROVED** [review]
